src/lib.rs

```diff
@@ -29,25 +34,70 @@ pub struct DstSender {
     inner: Weak<Inner>,
 }
 
+impl DstSender {
+    #[tracing::instrument(skip(self), name = "DstSender::send_endpoints", level = "info")]
+    pub fn send_endpoints(
+        &mut self,
+        dst: Dst,
+        endpoints: Endpoints,
+    ) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+        if let Some(sender) = self.endpoints.get(&dst) {
+            tracing::info!("Dst present");
+            sender.broadcast(endpoints)?;
+        } else {
+            tracing::info!("Dst non present");
+            if let Some(inner) = self.inner.upgrade() {
+                let (tx, rx) = watch::channel(endpoints);
+                self.endpoints.insert(dst.clone(), tx);
+                tokio::spawn(async move {
+                    inner.endpoints.write().await.insert(dst, rx);
+                });
```

Comment:
It occurs to me that I think we probably _don't_ want to use Tokio's `RwLock` here --- the lock is never actually held across an await. Using the standard library's synchronous `RwLock` would probably hurt performance less than spawning a new task and moving stuff into it just to acquire the lock. 

I know that we were using `tokio::sync::RwLock` here before this change, so we don't need to change it as part of this PR --- but it's probably good to do as a follow-up.

Reply:
For the short term, I think it would make more sense for `send_endpoints` and `delete_endpoints` to be `async fn`s than to spawn a new task for the hashmap manipulation...if we turned the loop over stream updates into a `while` loop, we could `await` them there.

Reply:
> For the short term, I think it would make more sense for send_endpoints and delete_endpoints to be async fns than to spawn a new task for the hashmap manipulation...if we turned the loop over stream updates into a while loop, we could await them there.

@hawkw  Thanks for the suggestion! That makes a lot of sense. That was one of my questions I wanted to ask, how to await in a map over a stream. I guess you cannot because you can only await within an async closure which is not what you pass into the map.

Reply:
> That was one of my questions I wanted to ask, how to await in a map over a stream. I guess you cannot because you can only await within an async closure which is not what you pass into the map.

FWIW, the `futures` crate's version of `StreamExt` has [`StreamExt::then`](https://docs.rs/futures/0.3.5/futures/stream/trait.StreamExt.html#method.then), which is essentially an async version of `map` --- it takes a closure returning an async block, which you can then await in. We could use that here rather than Tokio's version (which could also probably add a similar combinator). But, I think the best solution is probably just to make the lock not be async.